R/create_custom_check.R

```diff
@@ -0,0 +1,113 @@
+#' Create a custom validation check function template file.
+#'
+#' @param name Character string. Name of the custom check function. We recommend
+#' following the hubValidations package naming convention. For more details, consult the
+#' article on [writing custom check functions](
+#' https://hubverse-org.github.io/hubValidations/articles/writing-custom-fns.html).
+#' @param hub_path Character string. Path to the hub directory. Default is the
+#' current working directory.
+#' @param r_dir Character string. Path (relative to `hub_path`) to the directory
+#' the custom check function file will be written to. Default is `src/validations/R`
+#' which is the recommended directory for storing custom check functions.
+#' @param error Logical. If `TRUE`, the custom check function will return a
+#' `<error/check_error>` class object in the case of a failed check instead of the
+#' default `<error/check_failure>`.
```

Comment:
I easily forget the differences between `check_failure` and `check_error` and I think it would be helpful to reiterate the differences between the two. 

```suggestion
#' @param error Logical. Defaults to `FALSE`, which will return a 
#' `<error/check_failure>` class object in the case of a failed check. 
#' Set this to `TRUE` if your custom check function is required to pass for 
#' other checks to be performed; the custom check function will then return a
#' `<error/check_error>` class object in the case of a failed check instead.
```